regression/apps/fork/fork.s

```diff
@@ -22,13 +22,10 @@ _child:
     call    exit
 wait_child:
     mov     %rax, %rdi                  # child process id
-_loop:
     mov     $61, %rax                   # syscall number of wait4
     mov     $0, %rsi                    # exit status address
-    mov     $1, %rdx                    # WNOHANG
+    mov     $0, %rdx                    # wait option
     syscall
-    cmp     %rdi, %rax                  # The return value is the pid of child 
-    jne _loop
```

Comment:
The modification is ok. But comments should be added to emphasize that the new form is workaround. And it should be stored to the old form when the preemption is added. I recommend changing the file type to `.S` (not `.s`) so that the C preprocessor can be enabled. Then, you can use C macros to switch between the two forms easily.